test/tests/mortar/periodic_segmental_constraint/penalty_periodic_split.i

```diff
@@ -0,0 +1,226 @@
+  [sigma]
+    type = FunctionScalarAux
+    variable = sigma
+    function = '1 2 3'
+    execute_on = initial #timestep_end
+  []
+[]
+
+[Kernels]
+  [diff1]
+    type = Diffusion
+    variable = u
+    block = 1
+  []
+[]
+
+[Problem]
+  kernel_coverage_check = false
+  error_on_jacobian_nonzero_reallocation = true
+[]
+
+[BCs]
+  [fix_right]
+    type = DirichletBC
+    variable = u
+    boundary = pinned_node
+    value = 0
+  []
+[]
+
+[Constraints]
+  [mortarlr]
+    type = PenaltyEqualValueConstraint
+    primary_boundary = '12'
+    secondary_boundary = '14'
+    primary_subdomain = 'primary_right'
+    secondary_subdomain = 'secondary_left'
+    secondary_variable = u
+    correct_edge_dropping = true
+    penalty_value = 1.e2
+  []
+  [periodiclr]
+    type = PenaltyPeriodicSegmentalConstraint
+    primary_boundary = '12'
+    secondary_boundary = '14'
+    primary_subdomain = 'primary_right'
+    secondary_subdomain = 'secondary_left'
+    secondary_variable = u
+    epsilon = epsilon
+    sigma = sigma
+    correct_edge_dropping = true
+    penalty_value = 1.e2
+  []
+  [mortarbt]
+    type = PenaltyEqualValueConstraint
+    primary_boundary = '15'
+    secondary_boundary = '10'
+    primary_subdomain = 'primary_top'
+    secondary_subdomain = 'secondary_bottom'
+    secondary_variable = u
+    correct_edge_dropping = true
+    penalty_value = 1.e2
+  []
+  [periodicbt]
+    type = PenaltyPeriodicSegmentalConstraint
+    primary_boundary = '15'
+    secondary_boundary = '10'
+    primary_subdomain = 'primary_top'
+    secondary_subdomain = 'secondary_bottom'
+    secondary_variable = u
+    epsilon = epsilon
+    sigma = sigma
+    correct_edge_dropping = true
+    penalty_value = 1.e2
+  []
+  [mortarbf]
+    type = PenaltyEqualValueConstraint
+    primary_boundary = '13'
+    secondary_boundary = '11'
+    primary_subdomain = 'primary_front'
+    secondary_subdomain = 'secondary_back'
+    secondary_variable = u
+    correct_edge_dropping = true
+    penalty_value = 1.e2
+  []
+  [periodicbf]
+    type = PenaltyPeriodicSegmentalConstraint
+    primary_boundary = '13'
+    secondary_boundary = '11'
+    primary_subdomain = 'primary_front'
+    secondary_subdomain = 'secondary_back'
+    secondary_variable = u
+    epsilon = epsilon
+    sigma = sigma
+    correct_edge_dropping = true
+    penalty_value = 1.e2
+  []
+[]
+
+[Preconditioning]
+  active = 'FSP'
+
+  [./FSP]
+    type = FSP
+    topsplit = 'uv' # 'uv' should match the following block name
+    [./uv]
+      splitting = 'u v' # 'u' and 'v' are the names of subsolvers
+      splitting_type = additive
+    [../]
+    [./u]
+      vars = 'u'
+      petsc_options_iname = '-ksp_gmres_restart -pc_type -pc_hypre_type -pc_hypre_boomeramg_max_iter'
+      petsc_options_value = '  201               hypre    boomeramg      10'
+    [../]
+    [./v]
+      vars = 'epsilon'
+      petsc_options_iname = '-pc_type -ksp_type'
+      petsc_options_value = '     hypre  preonly'
```

Comment:
The default for `-pc_hypre_type` is `boomeramg`. I would prefer to explicitly specify it though

Reply:
Added